### PR TITLE
Added kernel information to data

### PR DESCRIPTION
**Pull request by ptxmac**
The main purpose of this is to have a way to detect WSL (windows subsystem for linux).

From what I can gather, the best way is currently to look in `/proc/sys/kernel/osrelease` and look for "microsoft" in the kernel name: https://github.com/microsoft/WSL/issues/423

### CODE REVIEW

**Comment by twpayne** [comment]
Thank you for this very nice contribution!

Please can you:
* Use build flags so this is only built when `GOOS=linux` (you can create a `getKernelInfo` function that returns nothing on other OSs, only set `data["kernel"]` if the return value is non-`nil`).
* Add documentation of the variable to `docs/REFERENCE.md`.

**Comment by ptxmac** [comment]
Sure, I'll get on that.

I based the implementation on `osrelease.go` since that's also a linux specific thing, but that isn't using build tags. Are there any reason for that?

**Comment by twpayne** [comment]
Thank you. `osrelease.go` not doing this is an oversight on my part. I'll fix it.

File: cmd/config.go [hunk 1]
```diff
@@ -320,6 +320,12 @@ func (c *Config) getDefaultData() (map[string]interface{}, error) {
 		return nil, err
 	}
 
+	kernelInfo, err := getKernelInfo(c.fs)
+	if err == nil {
+		data["kernel"] = kernelInfo
```
Comment (twpayne) [inline cmd/config.go]: Please only set `data["kernel"]` if `kernelInfo != nil`.

File: cmd/kernelinfo.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
+// +build linux
```
Comment (twpayne) [inline cmd/kernelinfo.go]: Please rename this file to `kernelinfo_linux.go` and remove the build tag.

File: cmd/kernelinfo_test.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
+// +build linux
```
Comment (twpayne) [inline cmd/kernelinfo_test.go]: Please rename this file to `kernelinfo_linux_test.go` and remove the build tag.

File: cmd/nokernelinfo.go [hunk 4]
```diff
@@ -0,0 +1,9 @@
+// +build !linux
```
Comment (twpayne) [inline cmd/nokernelinfo.go]: Please rename this file to `kernelinfo_notlinux.go` and keep the build tag.

File: docs/REFERENCE.md [hunk 5]
```diff
@@ -713,17 +713,18 @@ chezmoi's configuration file.
...
+| ----------------------- | ----------------------------------------------------------------------------------------------------------------------------- |
+| `.chezmoi.arch`         | Architecture, e.g. `amd64`, `arm`, etc. as returned by [runtime.GOARCH](https://godoc.org/runtime#pkg-constants).             |
+| `.chezmoi.fullHostname` | The full hostname of the machine chezmoi is running on.                                                                       |
+| `.chezmoi.group`        | The group of the user running chezmoi.                                                                                        |
+| `.chezmoi.homedir`      | The home directory of the user running chezmoi.                                                                               |
+| `.chezmoi.hostname`     | The hostname of the machine chezmoi is running on, up to the first `.`.                                                       |
+| `.chezmoi.kernel`       | Contains information from `/proc/sys/kernel`. Linux only, useful for detecting specific kernels (I.e. microsfot's WSL kernel) |
```
Comment (twpayne) [inline docs/REFERENCE.md]: Please fix the typos and capitalization and add a closing full stop: `(i.e. Microsoft's WSL kernel).`

File: cmd/config.go (hunk 1 above)
Comment (ptxmac) [reply cmd/config.go]: I guess I could also just return file not found from the no-op implementation

Comment (twpayne) [reply cmd/config.go]: Return a file not found error would have the same effect, but I think returning `nil, nil` more clearly signals the intent of the code.

File: cmd/nokernelinfo.go (hunk 4 above)
Comment (ptxmac) [reply cmd/nokernelinfo.go]: Man, go's build tags are getting messy :D - some can be expressed by file names, but only a subset...

File: cmd/config.go (hunk 1 above)
Comment (twpayne) [reply cmd/config.go]: This isn't quite right: if `err == nil` and `kernelInfo == nil` then the code ends up at line 327.

Comment (ptxmac) [reply cmd/config.go]: Yeah, I was a bit too quick with that fix

**Comment by twpayne** [comment]
Thank you very much for the great contribution!

**Pull request merged**